### PR TITLE
Torch dataloader and Lightning Trainer 

### PR DESCRIPTION
**Pull request by Sllambias**
Create a new Loader, Module and Dataset for PyTorch Lightning integration.
Still not completely finished but functional

### CODE REVIEW

File: yucca/training/trainers/YuccaLightningModule.py
Comment (asbjrnmunk) [inline yucca/training/trainers/YuccaLightningModule.py:13]: https://www.youtube.com/watch?v=4TYv2PhG89A

File: yucca/training/trainers/YuccaLightningWrapper.py
Comment (asbjrnmunk) [inline yucca/training/trainers/YuccaLightningWrapper.py:10]: I would prefer another name to avoid confusion with the lightning trainer. Maybe Yucca should not have a `trainer` concept anymore? Maybe something like `Operator` or `Manager`?

**Review by asbjrnmunk — CHANGES_REQUESTED** [review]
Overall looks really good! I am very happy that the code seems to fit this smoothly into the lightning framework. Hope it trains as fast 🤞. Most of these comments are very subjective and style-related. My suggestion would probably be to adapt something like [black](https://github.com/psf/black) and importantly to always be as [PEP 8](https://peps.python.org/pep-0008/) compliant as possible, such that style is consistent and not dependent on individual opinions.

File: yucca/training/trainers/YuccaLightningWrapper.py
Comment (Sllambias) [reply yucca/training/trainers/YuccaLightningWrapper.py:10]: Hmm, I like these actually

File: yucca/run/run_PLtraining.py
Comment (Sllambias) [reply yucca/run/run_PLtraining.py:7]: they actually do import lightning as L

**Comment by Sllambias** [comment]
Thanks Asbjørn. You are the best. I really appreciate everything you are doing for this project <3 I have updated the code based on your feedback.

File: yucca/training/trainers/YuccaLightningModule.py
Comment (asbjrnmunk) [inline yucca/training/trainers/YuccaLightningModule.py:80]: This line is tabs, the line above is spaces 😵‍💫

File: yucca/training/trainers/YuccaLightningWrapper.py
Comment (asbjrnmunk) [inline yucca/training/trainers/YuccaLightningWrapper.py:179]: Maybe call this `manager`

**Review by asbjrnmunk — APPROVED** [review]
LGTM! (when done) 🚀 (Would be nice to ensure all files are using spaces before merging to main).

**Pull request merged**
